src/frequenz/sdk/timeseries/logical_meter/_formula_steps.py

```diff
@@ -265,7 +262,18 @@ def apply(self, eval_stack: List[Optional[float]]) -> None:
         """
         if self._next_value is None:
             raise RuntimeError("No next value available to append.")
-        if self._next_value.value is None and self._nones_are_zeros:
-            eval_stack.append(0.0)
+
+        if len(eval_stack) and eval_stack[-1] is None:
+            if self._nones_are_zeros:
+                eval_stack[-1] = 0.0
+            else:
+                eval_stack[-1] = float("NaN")
```

Comment:
if individual steps don't insert `None`, then this check can go away.

Reply:
indeed, this was only added as a mechanism to handle the case where FormulaStep sets None